### PR TITLE
Replace Table.fetch() with __iter__()

### PR DESCRIPTION
**Pull request by Quasimewdo**
Previously, the user needs to call `Table.fetch()` explicitly to obtain the contents 
of `Table` object. This PR implements `Table.__iter__ `and `Table.__next__` which
allow the iteration through `Table` by simply calling `for row in Table`.

### CODE REVIEW

File: greenplumpython/table.py [hunk 1]
```diff
@@ -436,18 +436,18 @@ def db(self) -> Optional[db.Database]:
         """
         return self._db
 
-    def __len__(self) -> int:
+    def list(self) -> List[RealDictRow]:
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/table.py:439]: Yes, removed it

File: greenplumpython/table.py [hunk 2]
```diff
@@ -503,7 +493,47 @@ def _build_full_query(self) -> str:
             return self._query
         return "WITH " + ",".join(cte_list) + self._query
 
-    def fetch(self, is_all: bool = True) -> Iterable[Tuple[Any]]:
+    def __iter__(self) -> "Table":
+        if self._contents is not None:
+            self._n = 0
+            return self
+        assert self._db is not None
+        to_json_table = Table(
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/table.py:501]: Ok, no problem

File: greenplumpython/row.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    Represents a row of :class:`~table.Table`.
+    """
+
+    def __init__(self, contents: RealDictRow):
+        def detect_duplicate_keys(json_pairs: List[tuple[str, Any]]):
+            key_count = collections.Counter(k for k, _ in json_pairs)
+            duplicate_keys = ", ".join(k for k, v in key_count.items() if v > 1)
+
+            if len(duplicate_keys) > 0:
+                raise Exception("Duplicate key(s) found: {}".format(duplicate_keys))
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/row.py:22]: Will change it to `column_name`

File: greenplumpython/row.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+from psycopg2.extras import RealDictRow
+
+
+class Row:
+    """
+    Represents a row of :class:`~table.Table`.
+    """
+
+    def __init__(self, contents: RealDictRow):
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/row.py:16]: Yeah, will unify to `dict`

File: greenplumpython/table.py [hunk 5]
```diff
@@ -503,7 +500,61 @@ def _build_full_query(self) -> str:
...
+        Returns:
+            self
+        """
+
+        assert self._db is not None
+        result = self._fetch()
+        assert result is not None
+        self._contents = list(result)
+        self._n = 0
+        return self
```
Comment (yihong0618) [inline greenplumpython/table.py:555]: seems not test for this func and maybe we make it private like `_refresh` better?

File: tests/test_type.py [hunk 6]
```diff
@@ -39,6 +39,6 @@ def create_couple() -> Couple:
         return Couple()
 
     # FIXME : In this case, program will create twice Person type
-    #         when creating Couple type with different type_name
+    #         when creating Couple type with difƒrent type_name
```
Comment (yihong0618) [inline tests/test_type.py:42]: cool!

**Review by yihong0618 — APPROVED** [review]
LGTM

File: greenplumpython/table.py (hunk 5 above)
Comment (Quasimewdo) [reply greenplumpython/table.py:555]: I did add but it is disappeared, let me add it again

**Pull request merged**